src/coreclr/scripts/superpmi.py

```diff
@@ -1101,6 +1129,144 @@ async def run_crossgen2(print_prefix, assembly, self):
                 os.environ.update(old_env)
             ################################################################################################ end of "self.coreclr_args.crossgen2 is True"
 
+            ################################################################################################ Do collection using nativeaot
+            if self.coreclr_args.nativeaot is True:
+                logging.debug("Starting collection using nativeaot")
+
+                async def run_nativeaot(print_prefix, original_rsp_filepath, self):
+                    if not original_rsp_filepath.endswith(".ilc.rsp"):
+                        raise RuntimeError(f"Expected a '.ilc.rsp' file for nativeaot, but got {original_rsp_filepath}")
+
+                    rsp_filepath = os.path.join(self.temp_location, make_safe_filename("nativeaot_" + original_rsp_filepath) + ".rsp")
```

Comment:
It would be useful to have some comments here describing what is going to happen with the code that follows: the expected contents of the .rsp file, what transformations are required and why, etc.